### PR TITLE
Add accordion component

### LINKED ISSUE #38: add Accordion component

### PR DESCRIPTION
**Pull request by Donisius**
Closes #38

### CODE REVIEW

File: src/fragment-components/accordion/a-accordion-item.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	defaultComponentObj: {
+		type: 'accordionitem',
+		title: 'Accordion item',
+		disabled: false,
+		items: [{ type: 'text', text: 'Accordion item content' }]
+	},
+	image,
+	codeExport: {
+		angular: {
+			inputs: ({json}) => ``,
```
Comment (zvonimirfras) [inline src/fragment-components/accordion/a-accordion-item.tsx:111]: no inputs an outputs? 🤔

File: src/fragment-components/accordion/a-accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		type: 'accordion',
+		align: 'end',
+		size: 'md',
+		items: [
+			{
+				type: 'accordionitem',
+				title: 'Accordion item 1',
+				items: [{ type: 'text', text: 'Accordion item 1 content' }]
+			},
+			{
```
Comment (zvonimirfras) [inline src/fragment-components/accordion/a-accordion.tsx:89]: Do we need 3 items?

**Review by zvonimirfras — CHANGES_REQUESTED** [review]
It would be cool if the Adder supported all directions, configurable.

File: src/fragment-components/accordion/a-accordion-item.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import { ComponentCssClassSelector } from '../../components/css-class-selector';
+
+import image from '../../assets/component-icons/accordion-item.svg';
+import {
+	angularClassNamesFromComponentObj,
+	nameStringToVariableString,
+	reactClassNamesFromComponentObj
+} from '../../utils/fragment-tools';
+import { useFragment } from '../../context';
+import { Adder, getParentComponent, updatedState } from '../../components';
```
Comment (Akshat55) [inline src/fragment-components/accordion/a-accordion-item.tsx:17]:
```suggestion
import { 
    Adder, 
    getParentComponent, 
    updatedState 
} from '../../components';
```

Comment (Akshat55) [inline src/fragment-components/accordion/a-accordion-item.tsx:17]:
```suggestion
import { 
    Adder, 
    getParentComponent, 
    updatedState 
} from '../../components';
```

File: src/fragment-components/accordion/a-accordion-item.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	render: ({ componentObj, select, remove, selected, onDragOver, onDrop, renderComponents }) => <AAccordionItem
+		componentObj={componentObj}
+		select={select}
+		remove={remove}
+		onDragOver={onDragOver}
+		onDrop={onDrop}
+		selected={selected}>
+			{componentObj.items.map((child: any) => (
+				renderComponents(child)
+			))}
```
Comment (Akshat55) [inline src/fragment-components/accordion/a-accordion-item.tsx:102]:
```suggestion
			{componentObj.items.map((child: any) => renderComponents(child))}
```

File: src/fragment-components/accordion/a-accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	component: AAccordion,
+	styleUI: AAccordionStyleUI,
+	render: ({ componentObj, select, remove, selected, renderComponents }) => <AAccordion
+		componentObj={componentObj}
+		select={select}
+		remove={remove}
+		selected={selected}>
+		{componentObj.items.map((accordionItem: any) => (
+			renderComponents(accordionItem)
+		))}
```
Comment (Akshat55) [inline src/fragment-components/accordion/a-accordion.tsx:79]: Same as above - saves space 😛

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Review by Akshat55 — COMMENTED** [review]
Can approve right after that small lint fix

**Review by Akshat55 — CHANGES_REQUESTED** [review]
image
When you click on the text in accordion item, the component classifier box appears as if it is transparent. Need to update `z-index` to fix this.

**Review by Akshat55 — APPROVED** [review]

**Pull request merged**